sklift/viz/base.py

```diff
@@ -129,61 +131,86 @@ def plot_uplift_by_percentile(y_true, uplift, treatment, strategy, bins=10):
         raise ValueError(f'Number of bins = {bins} should be smaller than the length of y_true {n_samples}')
-    
-    if bins == 1:
-        warnings.warn(f'You will get the only one bin of {n_samples} samples'
-                      f' which is the length of y_true.' 
-                      f'\nPlease consider using uplift_at_k function instead',
-                      UserWarning)
-        
+
     rspns_rate_trmnt, var_trmnt = response_rate_by_percentile(y_true, uplift,
                                                               treatment, group='treatment',
                                                               strategy=strategy, bins=bins)
-    
+
     rspns_rate_ctrl, var_ctrl = response_rate_by_percentile(y_true, uplift,
                                                             treatment, group='control',
                                                             strategy=strategy, bins=bins)
 
     uplift_score, uplift_variance = np.subtract(rspns_rate_trmnt, rspns_rate_ctrl), np.add(var_trmnt, var_ctrl)
-    
+
     percentiles = [p * 100 / bins for p in range(1, bins + 1)]
-    
-    _, axes = plt.subplots(ncols=1, nrows=1, figsize=(8, 6))
-    
-    axes.errorbar(percentiles, uplift_score, yerr=np.sqrt(uplift_variance), 
-                  linewidth=2, color='red', label='uplift')
-    axes.errorbar(percentiles, rspns_rate_trmnt, yerr=np.sqrt(var_trmnt),
-                  linewidth=2, color='forestgreen', label='treatment\nresponse rate')
-    axes.errorbar(percentiles, rspns_rate_ctrl, yerr=np.sqrt(var_ctrl),
-                  linewidth=2, color='orange', label='control\nresponse rate')
-    axes.fill_between(percentiles, rspns_rate_ctrl, rspns_rate_trmnt, alpha=0.1, color='red')
-    
-    axes.set_xticks(percentiles)
-    axes.legend(loc='upper right')
-    axes.set_title('Uplift by percentile')
-    axes.set_xlabel('Percentile')
-    axes.set_ylabel('Uplift = treatment response rate - control response rate')
-    
+
+    if kind == 'line':
+        _, axes = plt.subplots(ncols=1, nrows=1, figsize=(8, 6))
+        axes.errorbar(percentiles, uplift_score, yerr=np.sqrt(uplift_variance),
+                      linewidth=2, color='red', label='uplift')
+        axes.errorbar(percentiles, rspns_rate_trmnt, yerr=np.sqrt(var_trmnt),
+                      linewidth=2, color='forestgreen', label='treatment\nresponse rate')
+        axes.errorbar(percentiles, rspns_rate_ctrl, yerr=np.sqrt(var_ctrl),
+                      linewidth=2, color='orange', label='control\nresponse rate')
+        axes.fill_between(percentiles, rspns_rate_ctrl, rspns_rate_trmnt, alpha=0.1, color='red')
+
+        if np.amin(uplift_score) < 0:
+            axes.axhline(y=0, color='black', linewidth=1)
+        axes.set_xticks(percentiles)
+        axes.legend(loc='upper right')
+        axes.set_title('Uplift by percentile')
+        axes.set_xlabel('Percentile')
+        axes.set_ylabel('Uplift = treatment response rate - control response rate')
```

Comment:
Duplicated code (lines 208-209, 211-212)
Maybe it shouldn't be in the "if" branch